lib/htmlbeautifier/html_parser.rb

```diff
@@ -4,7 +4,7 @@
 
 module HtmlBeautifier
   class HtmlParser < Parser
-    ELEMENT_CONTENT = %r{ (?:<%.*?%>|[^>])* }mx
+    ELEMENT_CONTENT = %r{ (?:<%.*?%>|data-action\s*=\s*"(?:[^"]*?->[^"]*?)"|[^>])* }mx
```

Comment:
Ideally this should be generalized rather being tied to `data-action`.

Reply:
Agreed. I simply couldn't figure out the regex magic to make it work without being this specific